ts-client/src/amm/index.ts

```diff
@@ -0,0 +1,863 @@
+    isSigner: boolean;
+  }> = [];
+  if ('stable' in poolState.curveType) {
+    if ('marinade' in poolState.curveType['stable'].depeg.depegType) {
+      accounts.push({
+        pubkey: CURVE_TYPE_ACCOUNTS.marinade,
+        isWritable: false,
+        isSigner: false,
+      });
+    }
+
+    if ('lido' in poolState.curveType['stable'].depeg.depegType) {
+      accounts.push({
+        pubkey: CURVE_TYPE_ACCOUNTS.solido,
+        isWritable: false,
+        isSigner: false,
+      });
+    }
+  }
+
+  return accounts;
+};
+
+const getPoolInfo = async ({
+  poolMint,
+  poolState,
+  program,
+  vaultA,
+  vaultB,
+  apyPda,
+}: {
+  poolMint: PublicKey;
+  poolState: PoolState;
+  program: AmmProgram;
+  vaultA: VaultImpl;
+  vaultB: VaultImpl;
+  apyPda: PublicKey;
+}) => {
+  const poolInfoInstruction = await program.methods
+    .getPoolInfo()
+    .accounts({
+      aVault: poolState.aVault,
+      aVaultLp: poolState.aVaultLp,
+      aVaultLpMint: vaultA.vaultState.lpMint,
+      bVault: poolState.bVault,
+      bVaultLp: poolState.bVaultLp,
+      bVaultLpMint: vaultB.vaultState.lpMint,
+      lpMint: poolState.lpMint,
+      pool: poolMint,
+      apy: apyPda,
+    })
+    .remainingAccounts(getRemainingAccounts(poolState))
+    .instruction();
+
+  const eventParser = new EventParser(program.programId, program.coder);
+  const poolInfoTransaction = new Transaction();
+  poolInfoTransaction.add(poolInfoInstruction);
+  poolInfoTransaction.feePayer = SIMULATION_USER;
+  const tx = await program.provider.connection.simulateTransaction(poolInfoTransaction);
+  const poolInfo = (await parseLogs(eventParser, tx.value.logs ?? [])) as PoolInformation;
+
+  return poolInfo;
+};
+
+export default class AmmImpl implements AmmImplementation {
+  public swapCurve: SwapCurve;
+  private vaultProgram: VaultProgram;
+  private opt: Opt = {
+    cluster: 'mainnet-beta',
+  };
+
+  private constructor(
+    private program: AmmProgram,
+    private apyPda: PublicKey,
+    private tokenInfos: Array<TokenInfo>,
+    public address: PublicKey,
+    public poolState: PoolState & PoolInformation & { lpSupply: BN },
+    public vaultA: VaultImpl,
+    public vaultB: VaultImpl,
+    private depegAccounts: Map<String, AccountInfo<Buffer>>,
+    opt: Opt,
+  ) {
+    this.vaultProgram = new Program<Vault>(VaultIdl, VAULT_PROGRAM_ID, this.program.provider);
+    this.opt = {
+      ...this.opt,
+      ...opt,
+    };
+
+    if ('stable' in this.poolState.curveType) {
+      const { amp, depeg, tokenMultiplier } = this.poolState.curveType['stable'];
+      this.swapCurve = new StableSwap(
+        amp.toNumber(),
+        tokenMultiplier,
+        depeg,
+        this.depegAccounts,
+        this.program.provider.connection,
+      );
+    } else {
+      this.swapCurve = new ConstantProductSwap();
+    }
+  }
+
+  public static async create(
+    connection: Connection,
+    pool: PublicKey,
+    opt?: {
+      cluster?: Cluster;
+      // programId?: string;
```

Comment:
Can we remove unused comments ?

Reply:
i think this will be useful later when we wanna support affiliate.

Reply:
nvm, i think i can just refer `vault_sdk` next time.